src/cc/CCIncludesMinerFee.cpp

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ * Copyright © 2021 The Komodo Core Developers.                               *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * SuperNET software, including this file may be copied, modified, propagated *
+ * or distributed except according to the terms contained in the LICENSE file *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ ******************************************************************************/
+#include "cc/CCIncludesMinerFee.h"
+
+/***
+ * Validate that the contract's conditions have been met
+ * @param eval the return value
+ * @param tx the transaction
+ * @param nIn not used
+ */
+bool IncludesMinerFee::Validate(Eval *eval, const CTransaction &tx, uint32_t nIn)
+{
+    CAmount total = tx.GetShieldedValueIn();
```

Comment:
Is this the correct way to get the total of vIn? I seem to remember reading a comment in some piece of code that the process has complications, so I'm wondering if this is correct.

Reply:
After further testing, I see that my assumption is correct. I cannot depend on GetShieldedValueIn() to give me the total of vIns.

Reply:
GetShieldedValueIn is for zcash transactions I guess. 
We usually iterate over vins, load vin txns and sum the txos values (I think there is also a more effective way using coin cache)